config/src/errors.rs

```diff
@@ -185,3 +185,46 @@ impl DsnpGraphError {
 		}
 	}
 }
+
+/// Macro to replicate Option<T>::ok_or, but logging if the returned
+/// Result is an Err variant.
+// (note: could have been implemented as a trait, but then the resulting log
+// event would not contain the correct file:line, since the std::file! macro records
+// the location only from an enclosing macro call)
+#[macro_export]
+macro_rules! ok_or_log {
+	($target:expr, $error:expr, $level:expr) => {{
+		$target.ok_or_else(|| {
+			log::log!($level, "{}", $error);
+			$error
+		})
+	}};
+	($target:expr, $error:expr, $level:expr, $context:expr) => {{
+		$target.ok_or_else(|| {
+			log::log!($level, "{}: {}", $context, $error);
+			$error
+		})
+	}};
+}
+
+/// Macro to log a Result::Err.
+// (note: could have been implemented as a trait, but then the resulting log
+// event would not contain the correct file:line, since the std::file! macro records
+// the location only from an enclosing macro call)
+#[macro_export]
+macro_rules! log_err {
+	($target:expr) => {{
+		let r = $target;
+		if let Some(err) = r.as_ref().err() {
+			log::error!("{}", err);
```

Comment:
lets do `{:?}`

Reply:
I wish there was a way to determine at runtime if a structure implements a trait, because `Display` looks a lot cleaner than `Debug` if there is an implementation.

Since we're currently only tagging `DsnpGraphResult<>` type of return values, and the associated `Error` type is `DsnpGraphError`, which uses the `thiserror` crate and thus automatically gets a `Display` implementation, how about we just leave it for now? If at any point things change it will cause a compile-time error (tagging the attribute on a Result type that does not implement the `Display` trait will fail at compile time). If that happens, we can change to using "{:?}" (though the same might be said if the variant doesn't implement the `Debug` trait...)